executor/src/main/java/com/redhat/service/smartevents/executor/ExecutorService.java

```diff
@@ -0,0 +1,77 @@
+package com.redhat.service.smartevents.executor;
+
+import java.net.URI;
+import java.util.UUID;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.CloudEventDeserializationException;
+import com.redhat.service.smartevents.infra.models.processors.ProcessorType;
+import com.redhat.service.smartevents.infra.utils.CloudEventUtils;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.builder.CloudEventBuilder;
+import io.cloudevents.jackson.JsonCloudEventData;
+
+@ApplicationScoped
+public class ExecutorService {
+
+    /**
+     * Channel used for receiving events.
+     */
+    public static final String EVENTS_IN_CHANNEL = "events-in";
+
+    private static final Logger LOG = LoggerFactory.getLogger(ExecutorService.class);
+    private static final String CLOUD_EVENT_SOURCE = "RHOSE";
+
+    @Inject
+    Executor executor;
+
+    @Inject
+    ObjectMapper mapper;
+
+    private Function<String, CloudEvent> toCloudEvent;
+
+    @PostConstruct
+    void init() {
```

Comment:
@kostola Any reason not to just in-line this in the `processEvent` method? Avoids anyone having to remember to call the `init` method. 

Not a blocking comment, just an observation.

Reply:
I inlined it in `processEvent` to simplify the code as you suggested